pandas/core/indexes/base.py

```diff
@@ -4028,6 +4028,11 @@ def get_indexer(
                 target, method=method, limit=limit, tolerance=tolerance
             )
 
+        if self._is_multi and self._contains_na_any_level:
```

Comment:
Id prefer fixing this on the engine level and not special casing

Reply:
ok, i'll take another look

Reply:
This is probably non trivial, not sure what's wrong there

Reply:
not seeing an obvious way to fix this on the engine level. If you look a few lines above this, categorical is already special cased to handle nans in a similar way. Are you ok with special casing for now?